py3status/modules/i3pystatus.py

```diff
@@ -0,0 +1,328 @@
+    then we just count the click as a single click. more than 2 clicks are also
+    viewed as a double click.
+
+    Additionally all buttons 6 and more use the unhandled click event but each
+    button is treated independently.
+    """
+
+    def __init__(self, parent, callbacks):
+        self.callbacks = callbacks
+        self.parent = parent
+        self.last_button = None
+        self.clicks = 0
+        self.click_time = getattr(
+            self.parent.module, 'multi_click_timeout', 0.25
+        )
+        self.timer = None
+
+    def trigger(self):
+        """
+        Actually trigger the event
+        """
+        if self.last_button:
+            button_index = min(self.last_button, len(self.callbacks)) - 1
+            click_style = 0 if self.clicks == 1 else 1
+            callback = self.callbacks[button_index][click_style]
+            if callback:
+                # we can do test for str as only python > 3 is supported
+                if isinstance(callback, str):
+                    # no parameters
+                    callback_name = callback
+                    callback_args = []
+                else:
+                    # has parameters
+                    callback_name = callback[0]
+                    callback_args = callback[1:]
+                callback_function = getattr(self.parent.module, callback_name)
+                callback_function(*callback_args)
+                # We want to ensure that the module has the latest output.  The
+                # best way is to call the run method of the module
+                self.parent.module.run()
+            else:
+                self.parent.py3.prevent_refresh()
+        self.last_button = None
+        self.clicks = 0
+
+    def event(self, button):
+        """
+        button has been clicked
+        """
+        # cancel any pending timer
+        if self.timer:
+            self.timer.cancel()
+        if self.last_button != button:
+            if self.last_button:
+                # new button clicked process the one before.
+                self.trigger()
+            self.clicks = 1
+        else:
+            self.clicks += 1
+        # do we accept double clicks on this button?
+        # if not then just process the click
+        button_index = min(button, len(self.callbacks)) - 1
+        if not self.callbacks[button_index][1]:
+            # set things up correctly for the trigger
+            self.clicks = 1
+            self.last_button = button
+            self.trigger()
+        else:
+            # set a timeout to trigger the click
+            # this will be cancelled if we click again before it runs
+            self.last_button = button
+            self.timer = Timer(self.click_time, self.trigger)
+            self.timer.start()
+
+
+SKIP_ATTRS = [
+    'on_click',
+    'run',
+    'post_config_hook',
+    'module',
+    'py3',
+]
+
+
+class Py3status:
+
```

Comment:
(Nit) Add missing empty `docstring` cause it's pretty...

Reply:
I think this is a waste of bytes :)

Reply:
still @tobes, please add them for consistency